dawn/src/dawn/CodeGen/Cuda/ASTStencilBody.cpp

```diff
@@ -125,14 +125,15 @@ void ASTStencilBody::derefIJCache(const std::shared_ptr<iir::FieldAccessExpr>& e
             " + (jblock - " + std::to_string(cacheProperties_.getOffsetBeginIJCache(accessID, 1)) +
             ")*" + std::to_string(cacheProperties_.getStride(accessID, 1, blockSizes_));
   }
-  DAWN_ASSERT(expr->getOffset()[2] == 0);
-
   auto offset = expr->getOffset();
+  DAWN_ASSERT(offset.verticalOffset() == 0);
+  auto const& hoffset = ast::offset_cast<ast::CartesianOffset const&>(offset.horizontalOffset());
```

Comment:
Please put a comment somewhere that this file (or CodeGen/Cuda in general) is only for cartesian grids.

Reply:
done in CudaCodeGen.h, which is the entry point.